functions/src/modules/SQR.functions.ts

```diff
@@ -209,6 +209,40 @@ export const restructureExternalSubmission = functions.database
   return 1;
 });
 
+const acceptedStatuses: string[] = ["Given", "WIP"];
+
+const addSubmissionWarnings = async (
+    currentSet,
+    { soundQualityReporting: { status, assignee } },
+    { cancellation: { notPreferredLanguage, audioProblem }, fileName, changed, completed, author, comments },
+    isFirstSubmission
+) => {
+  const listId = fileName.split("-")[0];
+  const warnings = [];
+  if (isFirstSubmission) warnings.push("This is the first submission by this devotee!");
+  if (notPreferredLanguage) warnings.push("The lecture is not in devotee's preferred language!");
+  if (audioProblem) warnings.push(`Unable to play or download the audio ${comments}`);
+  if (changed !== completed) warnings.push("This is an updated submission!");
+  if (assignee.emailAddress !== author.emailAddress) {
+    warnings.push(`File is alloted to another email id - ${assignee.emailAddress}`)
+  }
+  if (!(acceptedStatuses.includes(status))) {
+    warnings.push(`Status of file is ${status || "Spare"}`)
+  }
+  const response = (await db.ref(`/files/${listId}/${fileName}`).once("value")).val();
+  if (!response) warnings.push(`Audio file name ${fileName} is not found in the backend!`);
+  let shouldShowStatusWarning = true;
+  if (status === "Given") {
+    for (const allotment of currentSet) {
+      if (allotment.status === "Given") {
+        shouldShowStatusWarning = false;
+        break;
+      }
+    }
+    if (shouldShowStatusWarning) warnings.push("It's time to allot!");
```

Comment:
The trigger for this warning is if the current set has only one file in the “Given” status. I'm sure it's possible to simplify the  algorithm.